src/shared/utils/timers.ts

```diff
@@ -13,3 +15,56 @@ export const formatDate = (date: Date): string =>
       day: "numeric",
     })
     .toUpperCase()
+
+// This function calculates time remaining in the format: X days Y minutes remaining
+export const getTimeRemaining = (endDate: Date): string => {
+  const currentTimestamp = Date.now()
+  const endDateTimestamp = endDate.getTime()
+
+  const timeRemainingTimestamp = endDateTimestamp - currentTimestamp
+
+  const daysLeft = timeRemainingTimestamp / DAY
+  const hoursLeft = timeRemainingTimestamp / HOUR - Math.floor(daysLeft) * 24
+
+  // If less than 1 hour left, display "coming soon"
+  if (hoursLeft < 1) {
+    return XP_COMING_SOON_TEXT
+  }
+
+  let timeRemainingText = ""
+
+  if (daysLeft > 0) {
```

Comment:
we can skip showing `0 days` and just show hours
![image](https://github.com/tahowallet/dapp/assets/20949277/b9298625-b6f4-47b8-8c55-c4c7430d9e36)
```suggestion
  if (Math.floor(daysLeft) > 0) {
```